controller/instance_manager_controller.go

```diff
@@ -515,6 +520,130 @@ func (imc *InstanceManagerController) annotateCASafeToEvict(im *longhorn.Instanc
 	return nil
 }
 
+func (imc *InstanceManagerController) isSettingsSyncedToPod(im *longhorn.InstanceManager) (bool, error) {
+	if !im.DeletionTimestamp.IsZero() || im.Status.CurrentState != longhorn.InstanceManagerStateRunning {
+		return true, nil
+	}
+
+	for _, instance := range types.ConsolidateInstances(im.Status.InstanceEngines, im.Status.InstanceReplicas, nil) {
+		if instance.Status.State == longhorn.InstanceStateRunning || instance.Status.State == longhorn.InstanceStateStarting {
+			return true, nil
+		}
+	}
```

Comment:
If `!im.DeletionTimestamp.IsZero() || im.Status.CurrentState != longhorn.InstanceManagerStateRunning`, do we still need to check whether there are instances running in the pod?

Reply:
`!im.DeletionTimestamp.IsZero()` will be handled at the beginning of `syncInstanceManager`, so it can be removed here.
No, we don't need to check running instances if `CurrentState != longhorn.InstanceManagerStateRunnin` and here we will return to the caller immediately.